src/usr/share/gpgmailer/gpgmailer.py

```diff
@@ -89,8 +92,9 @@ def start_monitoring(self):
 
                     encrypted_message = self._build_encrypted_message(message_dict, loop_start_time)
 
+                    self.logger.debug(message_dict)
                     self.mailsender.sendmail(message_string=encrypted_message,
-                        recipients=self.valid_recipient_emails)
+                        recipients=message_dict['recipients'])
```

Comment:
This right here is correct, but when you encrypt the message, you should take the intersection of this variable and the self.valid_key_fingerprints variable. (Sorry, it wouldn't let me comment on those lines.)